### PR TITLE
Add action to automatically publish snapshots

### PR DESCRIPTION
**Pull request by jonkoops**
## Motivation
This will allow us to automatically publish a snapshot version on every commit on master, which can be used to in the main Keycloak project to include the latest changes from the Admin Console. For more information see @ssilvert's PR https://github.com/keycloak/keycloak/pull/8366

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+          java-version: 11
+          distribution: adopt
+          server-id: jboss-releases-repository
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+      - name: Publish package
+        run: mvn --batch-mode deploy
+        env:
+          MAVEN_USERNAME: ${{ secrets.JBOSS_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.JBOSS_TOKEN }}
```
Comment (jonkoops) [inline .github/workflows/maven.yml:24]: @ssilvert We're going to need these credentials in order to publish to the JBOSS repository, can you add them to the Github secrets?

File: keycloak-theme/pom.xml [hunk 2]
```diff
@@ -41,6 +41,14 @@
         </repository>
     </repositories>
 
+    <distributionManagement>
+        <repository>
+            <id>jboss-releases-repository</id>
+            <name>JBoss Releases Repository</name>
+            <url>https://repository.jboss.org/nexus/service/local/staging/deploy/maven2/</url>
+        </repository>
+    </distributionManagement>
```
Comment (stianst) [inline keycloak-theme/pom.xml:50]: Perhaps we should us GitHub Packages instead of Nexus?

Comment (jonkoops) [reply keycloak-theme/pom.xml:50]: We'd rather publish to the Nexus as that is already the configured repo for Keycloak.

File: .github/workflows/maven.yml (hunk 1 above)
Comment (jonkoops) [reply .github/workflows/maven.yml:25]: @stianst can you add these same credentials as Keycloak bot as an organization secret?

File: keycloak-theme/pom.xml (hunk 2 above)
Comment (jonkoops) [reply keycloak-theme/pom.xml:50]: After meeting we decided to publish snapshot versions to Github and release versions to Nexus.

File: .github/workflows/maven.yml (hunk 1 above)
Comment (jonkoops) [reply .github/workflows/maven.yml:25]: This is no longer required as we're going to be using Github as a repository.

File: .github/workflows/maven.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Publish package to JBOSS
+on: [pull_request]
+#on:
+#  push:
+#    branches: [master]
```
[the remark this answers is not in the document]
Comment (jonkoops) [reply .github/workflows/maven.yml]: Done

**Review by agagancarczyk — APPROVED** [review]

**Pull request merged**